src/handleLearnCliCommand.ts

```diff
@@ -5,8 +5,64 @@ import {
 	getLatestCommitHash,
 	isFileInGitDirectory,
 } from './gitCommands.js';
-import { spawnSync } from 'node:child_process';
-import { dirname } from 'node:path';
+import { exec, spawnSync } from 'node:child_process';
+import { dirname, extname } from 'node:path';
+import { Project, SourceFile, SyntaxKind } from 'ts-morph';
+import * as tmp from 'tmp';
+import fs from 'node:fs';
+
+const isJSorTS = (name: string) =>
+	name.startsWith('.ts') || name.startsWith('.js');
+const variableDeclarationRegex =
+	/(?:export\s+)?(?:var|let|const|type)\s+([a-zA-Z_$][a-zA-Z0-9_$]*)/;
+
+const getFileExtension = (filePath: string) => {
+	return extname(filePath).toLowerCase();
+};
+
+const getOldSourceFile = (
+	commitHash: string,
+	filePath: string,
+	fileExtension: string,
+): Promise<SourceFile | null> => {
+	return new Promise((resolve, reject) => {
+		if (!isJSorTS(fileExtension)) {
+			reject(null);
+		}
+		const gitShowCommand = `git show ${commitHash}:${filePath}`;
+
+		exec(gitShowCommand, (error, stdout) => {
+			if (error) {
+				reject(null);
+			}
+
+			const tmpFile = tmp.fileSync({ postfix: fileExtension });
+
+			try {
+				fs.writeFileSync(tmpFile.name, stdout, 'utf-8');
+
+				const project = new Project();
```

Comment:
project has an option to use an in-memory file system, you can see how I use it in codemod-registry :)

Reply:
amazing. addressed in https://github.com/intuita-inc/codemod-engine-node/pull/102/commits/80bdf0d5f21e30e26a6c18a1801f02cf19ec6bd6